jablib/src/main/java/org/jabref/model/metadata/MetaData.java

```diff
@@ -264,17 +260,36 @@ public void clearUserFileDirectory(String user) {
         postChange();
     }
 
-    public Optional<Path> getLatexFileDirectory(String user) {
-        return Optional.ofNullable(laTexFileDirectory.get(user));
+    public Optional<Path> getLatexFileDirectory(String userHostString) {
+        // First try to get the LaTeX file directory for the exact user-host
+        Path path = laTexFileDirectory.get(userHostString);
```

Comment:
rename variable to `latexFileDirectory` - with the capital `T` it really reads strange.